File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/Tests/Common/TextTests.cs
using NUnit.Framework;
using static Epam.JDI.Core.Settings.JDISettings;
using static JDI_UIWebTests.UIObjects.TestSite;
using Epam.JDI.Core.Interfaces.Common;
using static JDI_UIWebTests.Tests.Complex.CommonActionsData;
using Assert = JDI_Matchers.NUnit.Assert;

namespace JDI_UIWebTests.Tests.Common
{
    [TestFixture]
    public class TextTests
    {
        private IText _textItem = HomePage.Text;
        private string _expectedText = ("Lorem ipsum dolor sit amet, consectetur adipisicing elit,"
            + " sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
            + " Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris"
            + " nisi ut aliquip ex ea commodo consequat Duis aute irure dolor in"
            + " reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.").ToUpper();
        private string _regEx = ".* IPSUM DOLOR SIT AMET.*";
        private string _contains = "ENIM AD MINIM VENIAM, QUIS NOSTRUD";

        [SetUp]
        public void SetUp()
        {
            Logger.Info("Start test: " + TestContext.CurrentContext.Test.Name);
            HomePage.IsOpened();
            Logger.Info("Setup method finished");
        }

        [Test]
        public void GetTextTest() {
            Assert.AreEquals(_textItem.GetText, _expectedText);
        }

        [Test]
        public void GetValueTest()
        {
            Assert.AreEquals(_textItem.Value, _expectedText);
        }
        
        [Test]
        public void WaitMatchTest()
        {
            Assert.AreEquals(_textItem.WaitMatchText(_regEx), _expectedText);
        }

        [Test]
        public void WaitMatchTextParallelTest()
        {
            SupportPage.IsOpened();
            RunParallel(() => HomePage.IsOpened());
            Assert.AreEquals(_textItem.WaitMatchText(_regEx), _expectedText);
        }

        [Test]
        public void WaitText()
        {
            Assert.AreEquals(_textItem.WaitText(_contains), _expectedText);
        }

        [Test]
        public void WaitTextParallelTest()
        {
            SupportPage.IsOpened();
            RunParallel(() => HomePage.IsOpened());
            Assert.AreEquals(_textItem.WaitText(_contains), _expectedText);
        }

        [Test]
        public void SetAttributeTest()
        {
            string attributeName = "testAttr";
            string value = "testValue";
            _textItem.SetAttribute(attributeName, value);
            CheckText(() => _textItem.GetAttribute(attributeName), value);
        }
    }
}

File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/common/TextTests.java
package com.epam.jdi.uitests.testing.unittests.common;

import com.epam.jdi.uitests.core.interfaces.common.IText;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.enums.Preconditions;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.function.Supplier;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.core.settings.JDISettings.logger;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.SUPPORT_PAGE;
import static com.epam.web.matcher.testng.Assert.areEquals;
import static java.lang.String.format;

public class TextTests extends InitTests {
    private String elementType;
    private Preconditions page;
    private Supplier<? extends IText> textItem;
    private String expectedText;
    private String contains;
    private String regEx;

    public TextTests(String elementType, Preconditions page, Supplier<? extends IText> textItem, String expectedText,
                     String contains, String regEx) {
        this.elementType = elementType;
        this.page = page;
        this.textItem = textItem;
        this.expectedText = expectedText;
        this.contains = contains;
        this.regEx = regEx;
    }

    @BeforeMethod
    public void before(final Method method) {
        logger.info(format("=== Start test %s for %s ===", method.getName(), elementType));
        isInState(page, method);
    }

    @Test
    public void getTextTest() {
        areEquals(textItem.get().getText(), expectedText);
    }

    @Test
    public void getValueTest() throws Exception {
        areEquals(textItem.get().getValue(), expectedText);
    }

    @Test
    public void waitMatchText() {
        areEquals(textItem.get().waitMatchText(regEx), expectedText);
    }

    @Test
    public void waitMatchTextParallel() throws InterruptedException {
        isInState(SUPPORT_PAGE);

        Thread thread = new Thread(() -> {
            page.moveToAction();
        });
        thread.start();

        areEquals(textItem.get().waitMatchText(regEx), expectedText);

        thread.join();
    }

    @Test
    public void waitText() {
        areEquals(textItem.get().waitText(contains), expectedText);
    }

    @Test
    public void waitTextParallel() throws InterruptedException {
        isInState(SUPPORT_PAGE);

        Thread thread = new Thread(() -> {
            page.moveToAction();
        });
        thread.start();

        areEquals(textItem.get().waitText(contains), expectedText);

        thread.join();
    }
}
